server/src/server.rs

```diff
@@ -469,7 +467,6 @@ impl CubebServer {
             ServerMessage::ContextGetDeviceEnumeration(device_type) => context
                 .enumerate_devices(cubeb::DeviceType::from_bits_truncate(device_type))
                 .map(|devices| {
-                    self.devidmap.reset();
```

Comment:
It seems the amount of device in `devidmap` will only be increased, but never be decreased. Once device _D_ is enumerated, then _D_ will stay in `devidmap` until _cubeb context_ is destroyed, even _D_ is unplugged and `enumerate_devices` is called again? Is that right?

Reply:
Correct. That's the same contract libcubeb provides for cubeb_devid.

Reply:
I don't see that promise in [cubeb.h](https://github.com/kinetiknz/cubeb/blob/master/include/cubeb/cubeb.h), do you think that promise should be added?

But it's probably not that serious to guarantee the `devid` is workable, since we can always return an error when the device is invalid.

Reply:
That's what "across calls" is implying in the existing documentation, but maybe it could be clearer.  More details and the actual implementation are in https://github.com/kinetiknz/cubeb/issues/298 and https://github.com/kinetiknz/cubeb/pull/299.